### PR TITLE
Fix goaway

### PR DESCRIPTION
**Pull request by mcollina**
This PR changes how GOAWAY frames and user aborts are handled in HTTP/2.

In essence, we had a flaky test for a while that showed up locally on my machine after 2-3 hours of continuous runs. It's a bad race condition, which has been hard to find and a bad overlap between goaway, aborts, and manipulation of the `Client` state from the outside.

### CODE REVIEW

**Comment by mcollina** [comment]
I've opened this PR as draft with a lot of debug output, I will clean it up if this actually passes.

**Review by ronag — APPROVED** [review]

File: lib/dispatcher/client-h2.js [hunk 1]
```diff
@@ -463,26 +466,31 @@ function writeH2 (client, request) {
...
   stream.once('end', (err) => {
+    stream.removeAllListeners('data')
     // When state is null, it means we haven't consumed body and the stream still do not have
     // a state.
     // Present specially when using pipeline or stream
     if (stream.state?.state == null || stream.state.state < 6) {
-      request.onComplete([])
+      // The request was aborted
+      if (!request.aborted) {
+        request.onComplete([])
```
Comment (Uzlopak) [inline lib/dispatcher/client-h2.js:492]: I am just a little bit confused by the comment. Do you mean, The request was not aborted?

**Review by Uzlopak — APPROVED** [review]
I dont see any obvious issue
LGTM

**Comment by mcollina** [comment]
@Uzlopak PTAL

**Review by Uzlopak — APPROVED** [review]
LGTM :)

**Pull request merged**